platform/ios/scripts/generate-style-code.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const fs = require('fs');
+const ejs = require('ejs');
+const spec = require('mapbox-gl-style-spec').latest;
+
+global.camelize = function (str) {
+    return str.replace(/(?:^|-)(.)/g, function (_, x) {
+        return x.toUpperCase();
+    });
+}
+
+global.camelizeWithLeadingLowercase = function (str) {
+    return str.replace(/-(.)/g, function (_, x) {
+        return x.toUpperCase();
+    });
+}
+
+global.isObject = function (property) {
+    switch (property.type) {
+        case 'string':
+            return true;
+        case 'color':
+            return true;
+        case 'array':
+            return true;
+        case 'enum': // (enum will be converted to NSString hence isObject(enum) == true)
```

Comment:
If we’re going with the strongly typed design proposed in #5626, each enumeration type in the style specification would ideally be mapped to a C enum (`NS_ENUM`) rather than NSString.